docs/design/simple-rolling-update.md

```diff
@@ -0,0 +1,37 @@
+## Simple rolling update
+This is a lightweight design document for simple rolling update in ```kubectl```
+
+### Lightweight rollout
+Assume that we have a current replication controller named ```foo``` and it is running image ```image:v1```
+
+```kubectl rollingupdate rc foo --image=myimage:v2```
+
+Under the hood, this will do the following:
+   * Create a new replication controller named foo-next, identical to foo, except image is updated to the new image.
+   * Do the standard down/up one-by-one until foo has replicas zero and foo-next has replicas N
+   * delete rc foo
+   * create rc foo that is identical to foo-next
+   * delete foo-next
+
+Obviously there is a race here, where if you kill the client between delete foo, and creating the new version of 'foo' you might be surprised about what is there, but I think that's ok.
+See Recovery below
+
+There are the following flags to customize behavior:
+   * ```--controller-name``` sets the name of the _next_ controller
+   * ```--no-controller-rename``` prevents the renaming of the _next_ controller to the old name
+
+### Rollback
+Rollback is assumed to want to reverse a rollout in progress.  If the rollout has completed, then rollback is just a special case of rollout.
+
+```kubectl rollback rc foo```
```

Comment:
We need a special command for rollback of partially completed rollbacks.  In particular, we will need to reverse the ```desired-replicas``` annotions.

I agree that for rollbacks of updates that have fully completed, the rollback is actually just another rollout.

I would be ok with ```kubectl rolling-update rc foo abort``` instead.